lib/TPP/Dialect/Tpp/BufferizableOpInterfaceImpl.cpp

```diff
@@ -126,11 +128,53 @@ struct IdentityBufferizationInterface
   }
 };
 
+struct ZeroBufferizationInterface
+    : public BufferizableOpInterface::ExternalModel<ZeroBufferizationInterface,
+                                                    tpp::ZeroOp> {
+  bool bufferizesToMemoryRead(Operation *op, OpOperand &opOperand,
+                              const AnalysisState &state) const {
+    return bufferizesToMemoryReadUnaryImpl(op, opOperand, state);
+  }
+
+  bool bufferizesToMemoryWrite(Operation *op, OpOperand &opOperand,
+                               const AnalysisState &state) const {
+    return bufferizesToMemoryWriteUnaryImpl(op, opOperand, state);
+  }
+
+  AliasingOpResultList getAliasingOpResults(Operation *op, OpOperand &opOperand,
+                                            const AnalysisState &state) const {
+    return getAliasingOpResultsUnaryImpl(op, opOperand, state);
+  }
+
+  LogicalResult bufferize(Operation *op, RewriterBase &rewriter,
+                          const BufferizationOptions &options) const {
+    return bufferizeUnaryOp<tpp::ZeroOp>(op, rewriter, options);
+  }
+};
+
 //===----------------------------------------------------------------------===//
 // Binary
 //===----------------------------------------------------------------------===//
 
-static bool isConstantVal(Value val) { return matchPattern(val, m_Constant()); }
+static bool isConstantVal(Value val) {
+  if (auto toTensorOp = val.getDefiningOp<bufferization::ToTensorOp>()) {
+    Value memrefVal = toTensorOp.getMemref();
+    if (auto getGlobalOp = memrefVal.getDefiningOp<memref::GetGlobalOp>()) {
+      auto *symbolTableOp =
+          getGlobalOp->getParentWithTrait<OpTrait::SymbolTable>();
+      if (!symbolTableOp)
+        return false;
+      auto globalOp =
+          dyn_cast_or_null<memref::GlobalOp>(SymbolTable::lookupSymbolIn(
+              symbolTableOp, getGlobalOp.getNameAttr()));
+      if (!globalOp)
+        return false;
+      if (globalOp.getConstantInitValue())
+        return true;
+    }
+  }
+  return matchPattern(val, m_Constant());
```

Comment:
Nit: The shortest check should go first, so I'd change this to:
```
  auto toTensorOp = val.getDefiningOp<bufferization::ToTensorOp>();
  if (!tensorOp)
    return matchPattern(val, m_Constant());
  ...
```
https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code
"Aim to reduce indentation where possible when it doesn’t make it more difficult to understand the code."